pylisa/src/main/java/it/unive/pylisa/analysis/dataframes/transformation/operations/BooleanComparison.java

```diff
@@ -0,0 +1,84 @@
+package it.unive.pylisa.analysis.dataframes.transformation.operations;
+
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.program.SyntheticLocation;
+import it.unive.lisa.program.cfg.CodeLocation;
+import it.unive.pylisa.analysis.dataframes.constants.ConstantPropagation;
+import it.unive.pylisa.analysis.dataframes.transformation.operations.selection.Selection;
+import it.unive.pylisa.symbolic.operators.ComparisonOperator;
+
+public class BooleanComparison<S extends Selection<S>> extends DataframeOperation {
+
+    private S selection;
+    private ComparisonOperator op;
+    private ConstantPropagation value;
+
+    public BooleanComparison(CodeLocation where, S selection, ComparisonOperator op, ConstantPropagation value) {
+        super(where);
+        this.selection = selection;
+        this.op = op;
+        this.value = value;
+    }
+
+    @Override
+    protected boolean lessOrEqualSameOperation(DataframeOperation other) throws SemanticException {
+        BooleanComparison<?> o = (BooleanComparison<?>) other;
+        if (this.equals(o))
+            return true;
+        if (!this.selection.getClass().equals(o.selection.getClass()))
+            return false;
+        return selection.lessOrEqual((S) o.selection) && value.lessOrEqual(o.value);
```

Comment:
This cast to the parametric type `S` throws a warning here. Is there a better way to be doing this? @lucaneg